docs/topics/signals.rst

```diff
@@ -159,6 +208,13 @@ item_error
     :param failure: the exception raised
     :type failure: twisted.python.failure.Failure
 
+.. note::
```

Comment:
Why is this note under `item_error`? This seems relevant for all 3 item signals.

Maybe we could add subsections under “Built-in signals reference” that group related signals together (e.g. “Engine signals”, “Item signals”…) and have the note under “Item signals”, before the signal-specific sections.

Also, I think it may be better to remove the similar note from the source code example, to keep it shorter, and only have it here. I don’t think the note is completely necessary to illustrate what the example is meant to illustrate.

Reply:
Thanks for your review. I will add the following subsections 
1. Engine signals
2. Item signals
3. Spider signals
4. Request signals
5. Response signals 

under "Built-in signals reference". 

About the note in the source code example, I feel now that it is redundant as you said and will remove it.